AB#80874 Toggle show/hide archived records add records

guisoares1 opened the pull request:
# Description

When you go to form records you can toggle show/hide and it will duplicate some records.

## To fix

To resolve this, I clean the vector that stores the records that will be shown on the screen before starting the search process in the back-end.

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/80874)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [X] I went to the form records and toggle between show/hide and observed the behavior.

## Screenshots

https://github.com/ReliefApplications/ems-frontend/assets/55603259/fa87b142-c195-4036-a196-6b2bf9bfe09b

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

RenzoPrats reviewed (changes requested):
Hi @guisoares1, it seems to look fine but when we do some pagination and click to show active/archived records we have some issue like in this gif
![Peek 05-12-2023 17-07](https://github.com/ReliefApplications/ems-frontend/assets/56398308/bec40357-c124-4188-a3e1-1f78afcb5039)

guisoares1 commented:
@RenzoPrats I fixed this problem and removed the method's responsibility for clearing screen records and pagination. This process will only be done when you click on the show/hide archived records button.

RenzoPrats commented:
@guisoares1 
![Peek 05-12-2023 18-15](https://github.com/ReliefApplications/ems-frontend/assets/56398308/6dde6f5e-6c74-4a31-abeb-ec07497b9462)

guisoares1 commented:
Requested changes made!

RenzoPrats reviewed (approved):
Working fine, looks good to me!
